caps/types.go

```diff
@@ -20,66 +20,76 @@
 package caps
 
 import (
-	"encoding/json"
 	"fmt"
 )
 
 // Type describes a group of interchangeable capabilities with common features.
 // Types are managed centrally and act as a contract between system builders,
 // application developers and end users.
-type Type struct {
-	// Name is a key that identifies the capability type. It must be unique
-	// within the whole OS. The name forms a part of the stable system API.
-	Name string
-	// RequiredAttrs contains names of attributes that are required by
-	// capability of this type.
-	RequiredAttrs []string
+type Type interface {
+	// Unique and public name of this type.
+	Name() string
+	// Sanitize a capability (altering if necessary).
+	Sanitize(c *Capability) error
```

Comment:
The name changed because the method can now modify the capability and we wanted to make it explicit. I'm open to a better name, we picked sanitize in a quick discussion.